core/cosmosdb/cache-invalidator/README.md

```diff
@@ -0,0 +1,67 @@
+<!--
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+-->
+# OpenWhisk Cache Invalidator Service
+
+This service performs cache invalidation in OpenWhisk cluster to enable cache event propagation in multi region setups.
+
+## Design
+
+OpenWhisk cluster uses a Kafka topic `cacheInvalidation` to communicate changes to any cached entity. Messages on this
+topic are of the form
+
+```json
+{"instanceId":"controller0","key":{"mainId":"guest/hello"}}
+```
+
+When deploying multiple seprate cluster of OpenWhisk which do not share same Kafka instance we would need a way to
```

Comment:
When deploying multiple seprate cluster of OpenWhisk which do not share same Kafka instance we would need a way to propagate the cache change event across cluster. ->  When deploying OpenWhisk across multiple nodes which do not share a common Kafka instance, we  need a way to propagate the cache-change events across the cluster.